### PR TITLE
Fix errors in shell scripts.

### PR DESCRIPTION
**Pull request by uukhas**

### CODE REVIEW

**Comment by wkotlarski** [comment]
I'd really like to help but I'm not competent to do it. @Expander should have a look.

**Comment by uukhas** [comment]
@wkotlarski, the problem was only in some quotes. I am also fixing some warnings as well.
For example, POSIX does not support `local`, but we have a lot of them!

File: configure
Comment (Expander) [inline configure:3751]: I'm wondering why there is a warning on this line. Can you please just paste the warning here?

**Review by wkotlarski — APPROVED** [review]

**Comment by uukhas** [comment]
I have found, that the amount of warnings it too high. I will be following the strategy below:
* fix only errors
* fix warnings of type 1
* fix warnings of type 2
* ...

Then, in the end, there will be no obvious warnings and errors.
It might take some time :/

File: configure
Comment (uukhas) [reply configure]: `SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.`
and
`SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.`

Comment (uukhas) [reply configure]: I will fix this warning later, after all errors are fixed.

**Comment by uukhas** [comment]
@wkotlarski, in order to separate errors (few) and warnings (many) I would suggest to fix warnings in separate PR, after this one if merged to #385.

File: configure
Comment (Expander) [reply configure]: Thanks! I've just found that `-o` and `-a` are deprecated:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html

**Comment by Expander** [comment]
> I have found, that the amount of warnings it too high. I will be following the strategy below:
> 
>     * fix only errors
> 
>     * fix warnings of type 1
> 
>     * fix warnings of type 2
> 
>     * ...
> 
> 
> Then, in the end, there will be no obvious warnings and errors. It might take some time :/

That's a reasonable strategy.

**Pull request merged**
